packages/datastore/src/util.ts

```diff
@@ -681,3 +682,71 @@ export const isAWSIPAddress = (val: string): boolean => {
 		val
 	);
 };
+
+export function DeferredPromise(): void {
+	const self = this;
+	this.promise = new Promise((resolve, reject) => {
+		self.resolve = resolve;
+		self.reject = reject;
+	});
+}
```

Comment:
We can re-write this into a class like this:

```suggestion
class DeferredPromise {
	public promise: Promise<void>;
	public resolve: () => void;
	public reject: () => void;

	constructor() {
		const self = this;
		this.promise = new Promise((resolve, reject) => {
			self.resolve = resolve;
			self.reject = reject;
		});
	}
}
```